.github/workflows/build_windows_cmake.yml

```diff
@@ -0,0 +1,60 @@
+name: Build Windows CMake
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  job:
+    name: windows-cmake-build
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        build_config: ["Debug", "Release"]
+    env:
+      CMAKE_BUILD_DIR: ${{ github.workspace }}/vw/build
+      SOURCE_DIR: ${{ github.workspace }}/vw
+      VCPKG_ROOT: ${{ github.workspace }}/vcpkg
+      # 2020.11-1
+      VCPKG_REF: ec6fe06e8da05a8157dc8581fa96b36b571c1bd5
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: 'vw'
+          submodules: true
+      - uses: actions/checkout@v2
+        with:
+          path: 'vcpkg'
+          repository: 'microsoft/vcpkg'
+          ref: ${{ env.VCPKG_REF }}
+      - name: Restore vcpkg and build artifacts
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ env.VCPKG_ROOT }}/installed/
+            ${{ env.VCPKG_ROOT }}
+            !${{ env.VCPKG_ROOT }}/buildtrees
+            !${{ env.VCPKG_ROOT }}/packages
+            !${{ env.VCPKG_ROOT }}/downloads
+          key: |
+            ${{ env.VCPKG_REF }}-vcpkg-cache-invalidate-0
+      - uses: ilammy/msvc-dev-cmd@v1
+      - name: Bootstrap vcpkg
+        run: ${{ env.VCPKG_ROOT }}/bootstrap-vcpkg.bat
+      - name: Install vcpkg dependencies
+        run: ${{ env.VCPKG_ROOT }}/vcpkg.exe --triplet x64-windows install zlib boost-system boost-program-options boost-test boost-align boost-foreach boost-math flatbuffers
+      - name: Generate project files
+        run: |
+          cmake -S "${{ env.SOURCE_DIR }}" -B "${{ env.CMAKE_BUILD_DIR }}" -GNinja -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_ROOT }}/scripts/buildsystems/vcpkg.cmake" -DWARNINGS=Off -DUSE_LATEST_STD=On -DBUILD_FLATBUFFERS=On
```

Comment:
should we have warnings on?

Reply:
For some reason CMake add /W3 and we add /W4 and so it results in a message for every single file that we are overriding warnings

Reply:
Ah we just need to set the policy